### PR TITLE
BD-100

### PR DESCRIPTION
**Pull request by c247t**
Upstream Pull, Version Bump, Upstreaming von Änderungen aus dem Frontend

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -24,8 +24,8 @@
                 <artifactId>maven-compiler-plugin</artifactId>
                 <configuration>
                     <encoding>UTF-8</encoding>
-                    <source>11</source>
-                    <target>11</target>
+                    <source>17</source>
```
Comment (casid) [inline pom.xml:27]: Since this is a lib, do we want to force the latest Java LTS version?

**Review by casid — COMMENTED** [review]
We should enable github actions for this repository, so that we automatically run the build + tests on each commit/pr.

File: pom.xml (hunk 1 above)
Comment (c247t) [reply pom.xml:27]: In the upstreamed externalization code, there was one or more java17 switch expressions that I didn't want to backport to regular switch statements. Should not be a problem, though, if we decide that java11 shall be the baseline here.

**Comment by c247t** [comment]
I have enabled actions on the project, and the Java CI with Maven was already configured in the yaml.

**Review by casid — APPROVED** [review]

**Pull request merged**